### PR TITLE
Types

### PR DESCRIPTION
**Pull request by shubhbapna**
Exported types to fix https://github.com/kiegroup/github-action-build-chain/pull/319

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -10,6 +10,14 @@ import { readDefinitionFile } from "@bc-cr/reader";
 
 import { getMappedTarget, getMapping } from "@bc-cr/util/mapping";
 
+import { ArchiveArtifacts } from "@bc-cr/domain/archive-artifacts";
+import { Command, CommandLevel, BuildCommand, Build } from "@bc-cr/domain/build";
+import { DefinitionFile } from "@bc-cr/domain/definition-file";
+import { Dependency } from "@bc-cr/domain/dependencies";
+import { Mapping } from "@bc-cr/domain/mapping";
+import { Node } from "@bc-cr/domain/node";
+import { ReaderOpts } from "@bc-cr/domain/readerOptions";
```
Comment (Ginxo) [inline src/index.ts:19]: what is
```suggestion
import * as Domain from "@bc-cr/domain/";
```
 same strategy should be applied for the rest of domain submodules
and then 

```
export {
...
...
Domain
};
```

**Review by Ginxo — CHANGES_REQUESTED** [review]
@shubhbapna what about:
* `Pre` [1]
* `Post` [1]
* UrlPlaceholders [2]
* getBaseBranch [3]

check https://github.com/kiegroup/github-action-build-chain/actions/runs/3477112995/jobs/5812951240 to see different failures 

[1] https://github.com/kiegroup/github-action-build-chain/blob/8e692fb44a20148323dee6587f6053ca6a036f54/src/service/config/configuration-service.ts#L13 
[2] https://github.com/kiegroup/github-action-build-chain/blob/8e692fb44a20148323dee6587f6053ca6a036f54/src/service/config/definition-file-reader.ts#L19
[3] https://github.com/kiegroup/github-action-build-chain/blob/8e692fb44a20148323dee6587f6053ca6a036f54/src/service/checkout/checkout-service.ts#L5

**Comment by shubhbapna** [comment]
Ah I forgot about `Pre` and `Post`

We don't need `UrlPlaceholders` anymore, it is replaced by `ReaderOpts`.
Get `getBaseBranch` is now `getMappedTarget`

File: src/index.ts (hunk 1 above)
Comment (shubhbapna) [reply src/index.ts:19]: We will end up exporting all the domain objects even the ones which are mainly for internal use of the package. Is that okay?

Comment (shubhbapna) [reply src/index.ts:19]: I prefer the 1st option in that case. Will make the change

Comment (shubhbapna) [reply src/index.ts:19]: Nvm typescript complains about this line:
```typescript
import * as Domain from "@bc-cr/domain/";
```

**Pull request merged**